### PR TITLE
eslint: Enforce no-mixed-operators for boolean operators

### PR DESCRIPTION
**Pull request by 3nprob**
## Checklist

* [x] Tests written for new code (and old code if feasible)
* [x] Linter and other CI checks pass
* [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.md))

Type: Task

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

**Comment by 3nprob** [comment]
```
Signed-off-by: 3nprob <git@3n.anonaddy.com>
```

File: src/autodiscovery.ts [hunk 1]
```diff
@@ -410,38 +410,40 @@ export class AutoDiscovery {
...
-                        let reason = (err ? err.message : null) || "General failure";
-                        if (response && response.statusCode === 404) {
-                            action = AutoDiscoveryAction.IGNORE;
-                            reason = AutoDiscovery.ERROR_MISSING_WELLKNOWN;
-                        }
-                        resolve({ raw: {}, action: action, reason: reason, error: err });
-                        return;
+                (error, response, body) => {
+                    if (error || response?.statusCode < 200 || response?.statusCode >= 300) {
+                        const result = { error, raw: {} };
```
Comment (3nprob) [inline src/autodiscovery.ts:421]: `error` and `raw` can be moved into the conditional arms if readability trumps DRY here

**Review by weeman1337 — CHANGES_REQUESTED** [review]
Thank you for your contribution @3nprob .

Basically I am fine with introducing the rule. It matches [the new code style](https://github.com/vector-im/element-meta/pull/216/files#diff-97f46bd1ed0e6dfce7a7002e96f72ecd446b90c5eb5f48d687dc2d9df856290cR115).

But I think a better place for the rule would be the [Matrix ESLint Plugin](https://github.com/matrix-org/eslint-plugin-matrix-org).

I suggest closing this PR and opening a new one in the plugin repo.

**Comment by 3nprob** [comment]
Closing in favor of #2574

**Pull request closed without merging**